### PR TITLE
Add Acceleritas documentation

### PR DESCRIPTION
**Pull request by sethrj**
Fix some minor doxygen issues, add API documentation for the `accel` interface, and include the `demo-geant` example in the user manual.

### CODE REVIEW

File: doc/examples.rst [hunk 1]
```diff
@@ -8,4 +8,92 @@
...
 
-TODO: example applications using Celeritas.
+The following standalone codes demonstrate how to use Celeritas in an
+independent and Geant4-integrated context.
+
+Geant4 integration
+==================
+
+This lengthy example is a generic demonstration of using Celeritas to offload
+EM tracks from a Geant4 application. It is compatible with Geant4 10.7­11.0.
```
Comment (amandalund) [inline doc/examples.rst:18]:
```suggestion
EM tracks from a Geant4 application. It is compatible with Geant4 10.7­ and 11.0.
```

File: doc/examples.rst [hunk 2]
```diff
@@ -8,4 +8,92 @@
...
+
+.. literalinclude:: ../app/demo-geant-integration/TrackingAction.cc
+   :start-at: #include
+
+Sensitive Detectors
+-------------------
+
+The SD setup options in the ``DetectorConstruction`` constructor should be set
+as the union of requirements over all user Geant4 sensitive detectors. The SD
+The sensitive detector here is pretty generic and requires only the touchable,
```
Comment (amandalund) [inline doc/examples.rst:88]:
```suggestion
as the union of requirements over all user Geant4 sensitive detectors.
The sensitive detector here is pretty generic and requires only the touchable,
```

File: src/corecel/Assert.cc [hunk 3]
```diff
@@ -220,8 +220,7 @@ RuntimeError RuntimeError::from_geant_exception(const char* origin,
  * Construct an error message from a Geant4 exception.
  *
  * \param origin Usually the function that throws
- * \param code A computery error code
```
Comment (amandalund) [inline src/corecel/Assert.cc:223]: computery :)

**Review by amandalund — APPROVED** [review]
Looks good to me @sethrj, thanks for improving the documentation!

File: doc/examples.rst (hunk 1 above)
Comment (sethrj) [reply doc/examples.rst:18]: Whoops, the en dash got deleted by macvim...

File: src/corecel/Assert.cc (hunk 3 above)
Comment (sethrj) [reply src/corecel/Assert.cc:223]: In Geant4 they're arbitrary strings that look like fortran error codes: not human readable, not computer readable... just kinda "computery" 😉

**Review by whokion — APPROVED** [review]

**Pull request merged**